sdp-utils.c

```diff
@@ -0,0 +1,431 @@
+/*! \file    sdp-utils.c
+ * \author   Lorenzo Miniero <lorenzo@meetecho.com>
+ * \copyright GNU General Public License v3
+ * \brief    SDP utilities
+ * \details  Implementation of an internal SDP representation. Allows
+ * to parse SDP strings to an internal janus_sdp object, the manipulation
+ * of such object by playing with its properties, and a serialization
+ * to an SDP string that can be passed around. Since they don't have any
+ * core dependencies, these utilities can be used by plugins as well.
+ * 
+ * \ingroup core
+ * \ref core
+ */
+
+#include <string.h>
+ 
+#include "sdp-utils.h"
+#include "utils.h"
+#include "debug.h"
+
+#define JANUS_BUFSIZE	8192
+
+void janus_sdp_free(janus_sdp *sdp) {
+	if(!sdp)
+		return;
+	g_free(sdp->o_name);
+	g_free(sdp->o_addr);
+	g_free(sdp->s_name);
+	GList *temp = sdp->attributes;
+	while(temp) {
+		janus_sdp_attribute *a = (janus_sdp_attribute *)temp->data;
+		g_free(a->name);
+		g_free(a->value);
+		g_free(a);
+		temp = temp->next;
+	}
+	g_list_free(sdp->attributes);
+	temp = sdp->m_lines;
+	while(temp) {
+		janus_sdp_mline *m = (janus_sdp_mline *)temp->data;
+		g_free(m->type_str);
+		g_free(m->proto);
+		g_free(m->c_addr);
+		g_free(m->b_name);
+		g_list_free(m->ptypes);
+		GList *temp2 = m->attributes;
+		while(temp2) {
+			janus_sdp_attribute *a = (janus_sdp_attribute *)temp2->data;
+			g_free(a->name);
+			g_free(a->value);
+			g_free(a);
+			temp2 = temp2->next;
+		}
+		g_free(m);
+		temp = temp->next;
+	}
+	g_list_free(sdp->m_lines);
+	g_free(sdp);
+}
+
+janus_sdp_attribute *janus_sdp_attribute_create(const char *name, const char *value, ...) {
```

Comment:
I think this is what we're already doing, isn't it? https://github.com/meetecho/janus-gateway/blob/825cd22322ae7f4194d92a5bfcdbb3e4a83108cb/sdp-utils.h#L109